apps/api/src/app/widgets/usecases/mark-message-as-by-mark/mark-message-as-by-mark.usecase.ts

```diff
@@ -0,0 +1,116 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { MessageEntity, MessageRepository, SubscriberRepository, SubscriberEntity } from '@novu/dal';
+import { MarkMessagesAsEnum, WebSocketEventEnum } from '@novu/shared';
+import {
+  WebSocketsQueueService,
+  AnalyticsService,
+  InvalidateCacheService,
+  CachedEntity,
+  buildFeedKey,
+  buildMessageCountKey,
+  buildSubscriberKey,
+} from '@novu/application-generic';
+
+import { MarkMessageAsByMarkCommand } from './mark-message-as-by-mark.command';
+
+@Injectable()
+export class MarkMessageAsByMark {
+  constructor(
+    private invalidateCache: InvalidateCacheService,
+    private messageRepository: MessageRepository,
+    private webSocketsQueueService: WebSocketsQueueService,
+    private analyticsService: AnalyticsService,
+    private subscriberRepository: SubscriberRepository
+  ) {}
+
+  async execute(command: MarkMessageAsByMarkCommand): Promise<MessageEntity[]> {
+    await this.invalidateCache.invalidateQuery({
+      key: buildFeedKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    await this.invalidateCache.invalidateQuery({
+      key: buildMessageCountKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
```

Comment:
this invalidation logic should be after the subscriber validation

Reply:
good point, thanks